pkg/controllers/provisioning/scheduling/queue.go

```diff
@@ -96,6 +97,15 @@ func byCPUAndMemoryDescending(pods []*v1.Pod) func(i int, j int) bool {
 			return true
 		}
 
+		// anti-affinity pods should be sorted before normal pods
+		if affinityCmp := pod.PodAffinityCmp(lhsPod, rhsPod); affinityCmp != 0 {
```

Comment:
This seems like the right move, but I'm not sure how this breaks down in our bin-packing algorithm. From what I understand, this just sorts pods with affinity + tsc before others with the same exact pod requests.

Reply:
Yes, after testing this approach (there is a small test case in the previous section), scheduling the mutually exclusive pods further ahead helps to get a more balanced scheduling result

Reply:
With this approach, the cluster will be more stable (e.g., draining one node will not cause most pods to be rescheduled). I observed that Karpenter attempts to distribute the pods across all nodes:
[Scheduler Code](https://github.com/kubernetes-sigs/karpenter/blob/c4be45e04079f0d9313150b3ae7b5313132b0e36/pkg/controllers/provisioning/scheduling/scheduler.go#L259)

Reply:
cc @njtran @jonathan-innis , please take a look